crates/ibc/src/core/ics04_channel/error.rs

```diff
@@ -23,6 +24,8 @@ pub enum ChannelError {
     UnknownState { state: i32 },
     /// channel order type unknown: `{type_id}`
     UnknownOrderType { type_id: String },
+    /// invalid channel end error: `{0}`
```

Comment:
```suggestion
    /// invalid channel: `{0}`
```
I would also just take a `String` instead of a `ProtoError` to make the error more versatile.

Reply:
fcd8b791d29a82e838e7724c58ae4e804502bc29